### DISCUSSION TITLE
What exactly is listed in algorithms?

### DISCUSSION THREAD

**Discussion by KramNamez** [Yubico/libfido2#841] [Q&A] [answered]
I was recently trying to figure out what algorithms various fido key models would support, and noticed that `fido2-token -I` would sometimes list something under "algorithms" and sometimes not.

Do I just need to find the manual for each model, or is there a way to figure out if a model f.ex. supports ed25519 for use with SSH?

**Comment by LDVG** [comment, marked as answer]
Hi,

The "algorithms" field was added to CTAP 2.1 as an [optional field to the authenticatorGetInfo response structure](https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-20210615.html#getinfo-authenticatorgetinfo-response-structure). If the field is present, `fido2-token -I` will show it.

If the field is omitted, the only way to find out if an algorithm is supported is by trying to generate a credential.

**Comment by KramNamez** [comment]
Okay, so basically all my keys are being silly about this and either don't use the field or only show a weird excerpt of the actual list.

Thanks for the quick answer!
